src/core/pointcloud/qgspointcloudclassifiedrenderer.cpp

```diff
@@ -148,9 +154,10 @@ void QgsPointCloudClassifiedRenderer::renderBlock( const QgsPointCloudBlock *blo
       }
       else
       {
-        drawPoint( x, y, color, context );
+        const double size = pointSizes.value( attributeValue );
```

Comment:
I wonder if there's a better way to handle this then a map lookup for every point. Since we know that the attribute value will be integer and we know the max attribute value in advance, we could use an array for point sizes (and colors) and then just retrieve the values by index directly instead.

Reply:
This is a hash lookup with an int key, not a map, should be O(1) to read.
I tried this out of curiosity though :)
Using 
```cpp
QColor colors[255]
double sizes[255]
```
on a release build, a scene rendered in averagely 0.98s would benefit roughly 5ms.
I gained another 10ms by skipping re-creating the QColor in line 132 and using it's ref instead.
I'm not sure this is a significant gain...